main.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func mainMenu() (choice int) {
+	fmt.Println(`Choose your action:
+1) List all entries.
+2) Add new entry.
+3) Remove an entry by ID.
+4) Quit`)
+	fmt.Scanln(&choice)
+	return choice
+}
+
+func printHeader() {
+	fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+	fmt.Println("-------------------------------------------------------------------------")
+}
+
+func listEntries() {
+	var entries []Entry
+	f, err := os.Open("phonebook.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	sort.Slice(entries, func(i, j int) bool { return entries[i].LastName < entries[j].LastName })
+
+	printHeader()
+	for i, e := range entries {
+		fmt.Printf("%10d %20s %20s %20s\n", e.ID, e.LastName, e.FirstName, e.PhoneNumber)
+		if (i+1)%20 == 0 && i < len(entries)-1 { // Avoid the prompt for the last row.
+			fmt.Print("Press <ENTER> to continue...")
+			fmt.Scanln()
+			printHeader()
+		}
+	}
+}
+
+func newEntry() {
+	var entries []Entry
+	f, err := os.Open("phonebook.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+
+	var ID uint32
+
+	var id int // This piece of code finds new ID that will be greater than any existing ID.
```

Comment:
I will ask you the same questions as in the pull request this work is copied from:
- What's the difference between this `id` and `ID` declared above?
- Why would you need to type-cast?
- Why wouldn't you use `id` which contains the correct ID value? Instead you create an entry with 0-value ID
- Why would you later override that 0 with that other `id`?